### ISSUE TITLE
Support for java 8

### ISSUE THREAD

**Issue by geraldglennon** [hcoles/classycle-maven-plugin#8]
Hi,
we recently updated our project to java 8 however the plugin doesn't support lamda expressions and causes 

Execution verify of goal org.pitest:classycle-maven-plugin:0.4:check failed: java.io.EOFException -> 

When we removed Java 8 syntax it works as usual.

Have you any plans on adding support for JAVA 8. As we have to disable the plugin now :(

Thanks

**Comment by rnnr** [comment]
It's probably caused by classycle being compiled by old javac . I have recompiled version, published to maven central now, I'm going to do a pull request this week, depending on how quickly the previous pull request gets through.

**Comment by rnnr** [comment]
As this project seems abandoned, I've created a new version, Java 8 ready, based on the original source - it's available on maven central  as:

```
<groupId>org.bithill</groupId>
<artifactId>classycle-maven-plugin</artifactId>
<version>0.5</version>
```

Contributors are welcome : https://bitbucket.org/rostislav_matl/classycle-maven-plugin . 
I'm always busy but I'll try to process pull requests ASAP.

**Comment by sake92** [comment]
Hi, I had a similar error [here](https://github.com/sake92/sbt-hepek/issues/2). I dug into the source code of classycle and saw that [5 constant-pool tags](https://docs.oracle.com/javase/specs/jvms/se9/html/jvms-4.html#jvms-4.4-150) were missing.  
TL;DR, my fork can be found [here](https://github.com/sake92/hepek-classycle). It is released on maven. If you want I'll make a PR, since it is Hacktoberfest! 😄 

P.S. The support is very basic, it just parses the necessary bytes. The two **classes**, namely `MethodHandleConstant` and `InvokeDynamicConstant` aren't finished. If someone knows how to get the fields they reference, a PR is more than welcomed.
